File: native-ios/RandomTimer/Sources/Services/ProMonthlyContentMessaging.swift
import Foundation

/// User-facing copy for monthly Pro audio drop reminders.
enum ProMonthlyContentMessaging {
    struct Copy: Equatable {
        let title: String
        let body: String
    }

    static func monthLabel(releaseMonth: String) -> String {
        let trimmed = releaseMonth.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else {
            return fallbackMonthLabel()
        }

        let formatter = DateFormatter()
        formatter.locale = Locale(identifier: "en_US_POSIX")
        formatter.dateFormat = "yyyy-MM"
        guard let date = formatter.date(from: trimmed) else {
            return fallbackMonthLabel()
        }

        let display = DateFormatter()
        display.locale = Locale(identifier: "en_US")
        display.dateFormat = "MMMM yyyy"
        return display.string(from: date)
    }

    static func notificationCopy(releaseMonth: String) -> Copy {
        let label = monthLabel(releaseMonth: releaseMonth)
        return Copy(
            title: "New Audio Drops for \(label)",
            body: "Your Sound Arsenal has new tactical callouts. Open the app to train with the latest pack."
        )
    }

    private static func fallbackMonthLabel() -> String {
        let formatter = DateFormatter()
        formatter.locale = Locale(identifier: "en_US")
        formatter.dateFormat = "MMMM yyyy"
        return formatter.string(from: Date())
    }
}

File: native-android/app/src/main/java/com/iganapolsky/randomtimer/notifications/ProMonthlyContentMessaging.kt
package com.iganapolsky.randomtimer.notifications

import java.time.Month
import java.time.YearMonth
import java.time.format.DateTimeFormatter
import java.time.format.DateTimeParseException
import java.util.Locale

/**
 * User-facing copy for monthly Pro audio drop reminders.
 * Release month uses ISO `yyyy-MM` from the hosted runtime manifest.
 */
object ProMonthlyContentMessaging {
    data class Copy(
        val title: String,
        val body: String,
    )

    fun monthLabel(releaseMonth: String): String {
        val trimmed = releaseMonth.trim()
        if (trimmed.isEmpty()) {
            return fallbackMonthLabel()
        }
        return try {
            YearMonth.parse(trimmed, DateTimeFormatter.ofPattern("yyyy-MM"))
                .format(DateTimeFormatter.ofPattern("MMMM yyyy", Locale.US))
        } catch (_: DateTimeParseException) {
            fallbackMonthLabel()
        }
    }

    fun notificationCopy(releaseMonth: String): Copy {
        val label = monthLabel(releaseMonth)
        return Copy(
            title = "New Audio Drops for $label",
            body = "Your Sound Arsenal has new tactical callouts. Open the app to train with the latest pack.",
        )
    }

    private fun fallbackMonthLabel(): String {
        val now = YearMonth.now()
        val monthName = Month.of(now.monthValue).getDisplayName(java.time.format.TextStyle.FULL, Locale.US)
        return "$monthName ${now.year}"
    }
}
